src-cpp/rdkafkacpp.h

```diff
@@ -388,6 +388,7 @@ class Topic {
   static const int64_t OFFSET_END       = -1; /* Consume from end */
   static const int64_t OFFSET_STORED    = -1000; /* Consume from offset
                                                   * retrived from offset store*/
+  static const int64_t OFFSET_TAIL_BASE = -2000; /* internal: do not use */
```

Comment:
See comment below. Remove this and use the proper TAIL macro from rdkafka.h in the private method.